
Use freebsd/stdlib.h instead of bsd/stdlib.h, because we need devname
declaration as well.

--- src/sbin/swapon/swapon.c.old	2006-03-30 15:28:23.000000000 +0200
+++ src/sbin/swapon/swapon.c	2006-03-30 15:28:46.000000000 +0200
@@ -50,7 +50,7 @@
 #include <errno.h>
 #include <fstab.h>
 #include <stdio.h>
-#include <stdlib.h>
+#include <freebsd/stdlib.h>
 #include <string.h>
 #include <unistd.h>
 #include <fcntl.h>
@@ -70,6 +70,8 @@
 	int ch, doall;
 	int sflag = 0, lflag = 0, hflag = 0;
 
+	setprogname (argv[0]);
+
 	if ((ptr = strrchr(argv[0], '/')) == NULL)
 		ptr = argv[0];
 	if (strstr(ptr, "swapon"))
